### PR TITLE
Revert Default Model Config Setting for "default-space"

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

- Reverts model config default setting for "default-space".
- The old upgrade step to set it to "_default" is removed.
- A new upgrade step ensures that if unset or set to the old "_default" value, it is set to "".

**Edit:** It appears that the original addition of "default-space" as a model-config item was not released with 2.7-beta1, so checking for "_default" is not strictly necessary.

## QA steps

- Bootstrap 2.6.
- Build this patch and `juju upgrade-controller --build-agent`.
- Check that `juju model-config default-space` has the value "".

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Revert Model Config Space Default](https://trello.com/c/LYONATqU/1946-revert-model-config-space-default)

File: upgrades/steps_27.go [hunk 1]
```diff
@@ -77,17 +77,17 @@ func stateStepsFor27() []Step {
...
+			description: "replace space name in endpointBindingDoc bindings with an space ID",
 			targets:     []Target{DatabaseMaster},
 			run: func(context Context) error {
-				return context.State().AddDefaultSpaceSetting()
+				return context.State().ReplaceSpaceNameWithIDEndpointBindings()
 			},
 		},
 		&upgradeStep{
-			description: "replace space name in endpointBindingDoc bindings with an space ID",
+			description: `remove model config for default-space if set to "_default"`,
```
Comment (nammn) [inline upgrades/steps_27.go]: nit: Wouldn't the description be more along the line of remove model config + set default/initialize to ""?

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
```
13:20:46 Installing go-1.12 snap
13:21:52 error: cannot perform the following tasks:
13:21:52 - Setup snap "core" (7917) security profiles (cannot setup udev for snap "core": cannot reload udev rules: exit status 2
13:21:52 udev output:
13:21:52 )
13:21:52 - Setup snap "core" (7917) security profiles (cannot reload udev rules: exit status 2
13:21:52 udev output:
13:21:52 )
13:21:52 Makefile:151: recipe for target 'install-snap-dependencies' failed
13:21:52 make: *** [install-snap-dependencies] Error 1
13:21:52 + cleanup
13:21:52 + exit_code=2
```

$$merge$$

**Pull request merged**